### PR TITLE
Allow deletion of multiple allocations at once

### PR DESCRIPTION
**Pull request by ayan4m1**
Relates to #220 

I have added checkboxes to each item in the node allocation view. I have also added a Mass Actions dropdown patterned after the one in the file manager. You can check multiple allocations, then delete them all with one click (well, one click to activate one click to confirm).

[Here](https://imgur.com/XeL8Knf) are [some](https://imgur.com/GSFRo5b) screenshots.

I have tried to re-use code and style where possible.

Feedback is appreciated.

### CODE REVIEW

File: public/themes/pterodactyl/js/frontend/nodes/allocation.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
+class Allocation {
```
Comment (DaneEveritt) [inline public/themes/pterodactyl/js/frontend/nodes/allocation.js]: Can't use any of this unless you add it to babel to compile into ES2015. :(

Comment (ayan4m1) [reply public/themes/pterodactyl/js/frontend/nodes/allocation.js]: Should I do that or just inline the code (sans class) in `allocation.blade.php`?

Comment (DaneEveritt) [reply public/themes/pterodactyl/js/frontend/nodes/allocation.js]: Can you just write it using ES5?

Comment (ayan4m1) [reply public/themes/pterodactyl/js/frontend/nodes/allocation.js]: Should be done.

**Comment by DaneEveritt** [comment]
What happens with that popup when you select a ton of ports?

**Pull request merged**